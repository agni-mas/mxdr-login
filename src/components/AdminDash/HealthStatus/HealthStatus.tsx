import react, { FC, useEffect, useState } from 'react';
import GridItem from '../../layout/GridItem/GridItem';
import PieChartComponent from '../../global/PieChart/PieChart';
import Button from '@mui/material/Button';
import KeyboardArrowDownIcon from '@mui/icons-material/KeyboardArrowDown';
import { Skeleton } from '@mui/material';
import useFetchSecurityData from '../../../hooks/useFetchSecurityData';

interface HealthStatusProps { }

const HealthStatus: FC<HealthStatusProps> = () => {
    const [healthStatus, setHealthStatus] = useState([])

    const { data, loading } = useFetchSecurityData('HealthStatus.json')

    useEffect(() => {
        fetchData()
    }, [loading, data])

    const fetchData = () => {
        try {
            let rawData = (data) ?? []
            setHealthStatus(rawData)
        } catch (err) {
            console.error(err)
        }
    }

    return (
        <GridItem title="Health Status">
            <Button
                disableElevation
                endIcon={<KeyboardArrowDownIcon />}
                color='secondary'
                variant='outlined'
                sx={{ marginBottom: '40px' }}
            >
                Device Health
            </Button>
            {data ? (<PieChartComponent insideLabel="Total Health" data={healthStatus} type='analyst' />) : (<Skeleton variant="rectangular" animation="wave" height={200} />)}

            <Button color='secondary' variant="outlined" style={{ float: 'right', borderRadius: 28 }}>View Details</Button>
        </GridItem>
    )
}


export default HealthStatus;