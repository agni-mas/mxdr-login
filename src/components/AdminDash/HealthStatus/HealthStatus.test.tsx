import React from 'react';
import { render } from "@testing-library/react";
import HealthStatus from './HealthStatus';


it("renders <HealthStatus />", () => {
    const { asFragment } = render(<HealthStatus /> );
    expect(asFragment()).toMatchSnapshot();
});
