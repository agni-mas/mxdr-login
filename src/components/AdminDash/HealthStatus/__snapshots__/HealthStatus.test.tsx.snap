// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders <HealthStatus /> 1`] = `
<DocumentFragment>
  <section
    class="item"
  >
    <h2
      class="grid-item-title"
    >
      Health Status
    </h2>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedSecondary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-disableElevation MuiButton-root MuiButton-outlined MuiButton-outlinedSecondary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-disableElevation css-jqgntc-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      Device Health
      <span
        class="MuiButton-endIcon MuiButton-iconSizeMedium css-9tj150-MuiButton-endIcon"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="KeyboardArrowDownIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M7.41 8.59 12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z"
          />
        </svg>
      </span>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
    <span
      class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-wave css-1kcnzpi-MuiSkeleton-root"
      style="height: 200px;"
    />
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedSecondary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedSecondary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-wnpvqa-MuiButtonBase-root-MuiButton-root"
      style="float: right; border-radius: 28px;"
      tabindex="0"
      type="button"
    >
      View Details
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </section>
</DocumentFragment>
`;
